sdk/python/sawtooth_sdk/client/sdk_client.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+import json
+import urllib.request as urllib
+from urllib.error import URLError, HTTPError
+from http.client import RemoteDisconnected
+
+from sawtooth_sdk.client.exceptions import CliException
+
+
+def _submit_request(url_or_request):
+    try:
+        result = urllib.urlopen(url_or_request)
+        return (result.status, json.loads(result.read().decode()))
+    except HTTPError as e:
+        return (e.code, e.msg)
+    except RemoteDisconnected as e:
+        raise CliException(e)
+    except URLError as e:
+        raise CliException(e)
+
+
+def post_batches(url, batches):
```

Comment:
This throws away the error code (from a programatic perspective), so any code using post_batches is limited to success/failure.  The error message may or may not be formatted in a way that is good for users of the SDK.  These seems like a specific convenience wrapper that should live in app code (intkey, whatever) and not in the SDK.

The SDK is not CLI-specific, so CliException isn't appropriate.  This seems more like a BatchSubmissionException.

If we are going to abstract the use of the REST API at all, seems like we should do it in a way that provides the ability for a client to select between rest api and interconnect while keeping the same API.  That might mean rethinking the API of stream.